### ISSUE TITLE
Reduce chunk size after minification

### ISSUE THREAD

**Issue by dlwhitehurst** [virtualyou/app#16]
Output from build:

```bash
david@INTEL-NUC:~/development/virtualyou/app$ npm run build

> app@0.2.5 build
> tsc && vite build

vite v4.5.0 building for production...
✓ 1764 modules transformed.
dist/index.html                     0.42 kB │ gzip:   0.29 kB
dist/assets/index-d702c405.css    257.84 kB │ gzip:  35.09 kB
dist/assets/index-7546bb98.js   1,312.89 kB │ gzip: 336.80 kB

(!) Some chunks are larger than 500 kBs after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
✓ built in 14.55s
```

Fix please

**Comment by dlwhitehurst** [comment]
<< BING_CHAT
To use Rollup with TypeScript and manualChunks, you need to create a configuration file for Rollup and add the necessary plugins. Here are the steps to follow:

1. Install the required packages:

```
npm install --save-dev rollup rollup-plugin-typescript2 @rollup/plugin-commonjs @rollup/plugin-node-resolve
```

2. Create a `rollup.config.js` file in the root directory of your project and add the following code:

```javascript
import typescript from 'rollup-plugin-typescript2';
import commonjs from '@rollup/plugin-commonjs';
import resolve from '@rollup/plugin-node-resolve';

export default {
  input: 'src/index.ts',
  output: {
    dir: 'dist',
    format: 'esm',
  },
  plugins: [
    resolve(),
    commonjs(),
    typescript({
      tsconfig: './tsconfig.json',
    }),
  ],
  manualChunks(id) {
    if (id.includes('node_modules')) {
      return 'vendor';
    }
  },
};
```

[This configuration file tells Rollup to use TypeScript to compile your code, resolve and bundle external dependencies, and create a separate vendor chunk for external dependencies ](https://www.thisdot.co/blog/how-to-setup-a-typescript-project-using-rollup-js)¹.

3. Update your `package.json` file to include the following scripts:

```json
{
  "scripts": {
    "build": "rollup -c"
  }
}
```

4. Run the build command to generate the bundled files:

```
npm run build
```

I hope this helps! Let me know if you have any other questions..

Source: Conversation with Bing, [1](https://www.thisdot.co/blog/how-to-setup-a-typescript-project-using-rollup-js)2/5/2023
(1) How to Setup a TypeScript project using Rollup.js. https://www.thisdot.co/blog/how-to-setup-a-typescript-project-using-rollup-js.
(2) @rollup/plugin-typescript - npm. https://www.npmjs.com/package/@rollup/plugin-typescript.
(3) webpack - Generate typescript definition files using rollup - Stack .... https://stackoverflow.com/questions/41329876/generate-typescript-definition-files-using-rollup.
(4) Does rollup support typescript in rollup config file?. https://stackoverflow.com/questions/54711437/does-rollup-support-typescript-in-rollup-config-file.
(5) Using Rollup to package a library for TypeScript and JavaScript. https://blog.logrocket.com/using-rollup-package-library-typescript-javascript/.
(6) Component library setup with React, TypeScript and Rollup. https://dev.to/siddharthvenkatesh/component-library-setup-with-react-typescript-and-rollup-onj.
(7) Building for Production | Vite. https://vitejs.dev/guide/build.
(8) Configuration Options | Rollup. https://rollupjs.org/configuration-options/.
(9) Building for Production | Vite. https://v2.vitejs.dev/guide/build.
(10) reactjs - How to bundle react app with rollup - Stack Overflow. https://stackoverflow.com/questions/51291553/how-to-bundle-react-app-with-rollup.
(11) undefined. https://github.com/luixaviles/typescript-rollup.git.
(12) undefined. https://github.com/luixaviles/typescript-rollup/issues.
(13) undefined. https://github.com/luixaviles/typescript-rollup.
(14) Command Line Interface | Rollup. https://rollupjs.org/command-line-interface/.
(15) TypeScript library tips: Rollup your types! | by Martin Hochel - Medium. https://medium.com/@martin_hotell/typescript-library-tips-rollup-your-types-995153cc81c7.
(16) en.wikipedia.org. https://en.wikipedia.org/wiki/TypeScript.
BING_CHAT

**Comment by dlwhitehurst** [comment]
I fixed this @v0.2.6-beast with the following config or `vite.config.ts`
```
/// <reference types="vite/client" />
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'

// https://vitejs.dev/config/

export default defineConfig({
        plugins: [
            react(),
        ],
        server: {
            port: 3000,
            proxy: {
                '/userauth': 'http://localhost:3001', 
                '/personal': 'http://localhost:3002',
                '/medical': 'http://localhost:3003',
                '/financial': 'http://localhost:3004',
                '/administration': 'http://localhost:3005'
            }
        },
        build: {
            chunkSizeWarningLimit: 2000,
            rollupOptions: {
                output: {
                    manualChunks(id) {
                        if (id.includes('node_modules')) {
                            return id.toString().split('node_modules/')[1].split('/')[0].toString();
                        }
                    },
                },
            }
        }
})

```
